packages/remix/src/build.ts

```diff
@@ -575,3 +575,23 @@ async function chdirAndReadConfig(dir: string) {
   }
   return remixConfig;
 }
+
+async function copyEntrypointFile(
+  source: string,
+  dest: string,
+  rootDir: string
+) {
+  try {
+    await fs.copyFile(source, dest);
```

Comment:
`fs.copyFile()` will throw an `ENOENT` if the destination directory does not exist. I suggest creating the dest directory and see if that fixes the problem.

```suggestion
        await fs.mkdir(dirname(dest), { recursive: true });
        await fs.copyFile(source, dest);
```

Reply:
I'm expecting the `ENOENT` (see the catch block). If we make the dir implicitly then that would paper over something wrong in the Remix build process (the projects server bundles would not exist, so the deployment is gonna fail anyways).

The point of this PR is to show a better error than `ENOENT` specifically for that case.

Reply:
So you are saying there is no race conditions, the destination directory should always exist, and this message should never be displayed because things should just work in which you really are expecting `ENOENT` if/when something real bad happens.

Reply:
Correct, under normal circumstances, the `ENOENT` shouldn't happen. If it does, then it's probably user error like the case I described in the OP about a project using Turbo but not having the "build" directory defined as an output that needs to be cached.